lib/dogma/rules/hard_tabs.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Dogma.Rules.HardTabs do
+  @moduledoc """
+  Requires that all indentation is done using spaces rather than hard tabs.
+
+  So the following would be invalid:
+
+  def something do
+  \t:body
+  end
+  """
+  @behaviour Dogma.Rule
+
+  @indenting_tab_pattern ~r/^\t+.+/
```

Comment:
Do we want to check that the line starts with a tab, or that they are in the source at all?